components/feature/readerview/src/main/java/mozilla/components/feature/readerview/internal/ReaderViewConfig.kt

```diff
@@ -0,0 +1,102 @@
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.COLOR_SCHEME_KEY
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_SIZE_DEFAULT
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_SIZE_KEY
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_TYPE_KEY
+import org.json.JSONObject
+import java.util.Locale
+
+/**
+ * Class to store the user configuration for reader view.
+ * Values are initialized lazily, and will send a message when they are changed.
+ */
+internal class ReaderViewConfig(
+    private val prefs: SharedPreferences,
+    private val sendConfigMessage: (JSONObject) -> Unit
+) {
+
+    private var colorSchemeCache: ReaderViewFeature.ColorScheme? = null
+    private var fontTypeCache: ReaderViewFeature.FontType? = null
+    private var fontSizeCache: Int? = null
+
+    var colorScheme: ReaderViewFeature.ColorScheme
+        get() {
+            if (colorSchemeCache == null) {
+                colorSchemeCache = getEnum(COLOR_SCHEME_KEY, ReaderViewFeature.ColorScheme.LIGHT)
+            }
+            return colorSchemeCache!!
+        }
+        set(value) {
+            colorSchemeCache = value
+            setEnumAndSend(COLOR_SCHEME_KEY, ACTION_SET_COLOR_SCHEME, value)
+        }
+
+    var fontType: ReaderViewFeature.FontType
+        get() {
+            if (fontTypeCache == null) {
+                fontTypeCache = getEnum(FONT_TYPE_KEY, ReaderViewFeature.FontType.SERIF)
+            }
+            return fontTypeCache!!
+        }
+        set(value) {
+            fontTypeCache = value
+            setEnumAndSend(FONT_TYPE_KEY, ACTION_SET_FONT_TYPE, value)
+        }
+
+    var fontSize: Int
+        get() {
+            if (fontSizeCache == null) {
+                fontSizeCache = prefs.getInt(FONT_SIZE_KEY, FONT_SIZE_DEFAULT)
+            }
+            return fontSizeCache!!
+        }
+        set(value) {
+            val diff = value - fontSize
+            fontSizeCache = value
+            prefs.edit().putInt(FONT_SIZE_KEY, value).apply()
+            sendMessage(ACTION_CHANGE_FONT_SIZE) { put(ACTION_VALUE, diff) }
+        }
+
+    fun createShowReaderMessage(): JSONObject {
+        val configJson = JSONObject()
+            .put(ACTION_VALUE_SHOW_FONT_SIZE, fontSize)
+            .put(ACTION_VALUE_SHOW_FONT_TYPE, fontType.name.toLowerCase(Locale.ROOT))
+            .put(ACTION_VALUE_SHOW_COLOR_SCHEME, colorScheme.name.toLowerCase(Locale.ROOT))
+
+        return JSONObject()
+            .put(ACTION_MESSAGE_KEY, ACTION_SHOW)
+            .put(ACTION_VALUE, configJson)
+    }
+
+    private inline fun <reified T : Enum<T>> getEnum(key: String, default: T): T {
+        val enumName = prefs.getString(key, default.name) ?: default.name
+        return enumValueOf(enumName)
+    }
+
+    private fun <T : Enum<T>> setEnumAndSend(key: String, action: String, value: T) {
```

Comment:
nit: I think it's much easier to read if we just inline those two calls again and remove `setEnumAndSend`. We only have two of those and `setEnumAndSend` doesn't make this easier to read...